jenkins/bootstrap.py

```diff
@@ -327,7 +315,24 @@ def finish(gsutil, paths, success, artifacts, build, version, repo):
     if version:
         data['job-version'] = version
         data['version'] = version  # TODO(fejta): retire
-    gsutil.upload_json(paths.finished, data)  # Raise if this fails
+    finished_ex = None
+    try:
+        gsutil.upload_json(paths.finished, data)
+    except subprocess.CalledProcessError as finished_ex:
+        logging.warning('Failed to upload %s', paths.finished)
+
+    # Upload the latest build for the job.
+    # Do this last, since other tools expect the rest of the data to be
+    # published when this file is created.
+    for path in {paths.latest, paths.pr_latest}:
```

Comment:
I'm not technically sure this is necessary, do we really want to update the latest build if we fail to upload finished.json?

When would we want to do this?